esmf_regrid/experimental/io.py

```diff
@@ -0,0 +1,150 @@
+        for coord in src_mesh.to_MeshCoords("face"):
+            src_cube.add_aux_coord(coord, 0)
+
+        tgt_grid = (rg.grid_y, rg.grid_x)
+        tgt_shape = [len(coord.points) for coord in tgt_grid]
+        tgt_data = np.zeros(tgt_shape)
+        tgt_cube = Cube(tgt_data, long_name=tgt_name)
+        tgt_cube.add_dim_coord(tgt_grid[0], 0)
+        tgt_cube.add_dim_coord(tgt_grid[1], 1)
+    else:
+        msg = (
+            f"Expected a regridder of type `GridToMeshESMFRegridder` or "
+            f"`MeshToGridESMFRegridder`, got type {type(rg)}."
+        )
+        raise TypeError(msg)
+
+    metadata_name = "regridder weights and metadata"
+
+    weight_matrix = rg.regridder.weight_matrix
+    reformatted_weight_matrix = weight_matrix.tocoo()
+    weight_data = reformatted_weight_matrix.data
+    weight_rows = reformatted_weight_matrix.row
+    weight_cols = reformatted_weight_matrix.col
+    weight_shape = reformatted_weight_matrix.shape
+
+    mdtol = rg.mdtol
+    attributes = {
+        "regridder type": regridder_type,
+        "mdtol": mdtol,
+        "weights shape": weight_shape,
+    }
+
+    metadata_cube = Cube(weight_data, long_name=metadata_name, attributes=attributes)
+    row_name = "weight matrix rows"
+    row_coord = AuxCoord(weight_rows, long_name=row_name)
+    col_name = "weight matrix columns"
+    col_coord = AuxCoord(weight_cols, long_name=col_name)
+    metadata_cube.add_aux_coord(row_coord, 0)
+    metadata_cube.add_aux_coord(col_coord, 0)
+
+    # Avoid saving bug by placing the mesh cube second.
+    # TODO: simplify this when this bug is fixed in iris.
+    if regridder_type == "grid to mesh":
+        cube_list = CubeList([src_cube, tgt_cube, metadata_cube])
+    elif regridder_type == "mesh to grid":
+        cube_list = CubeList([tgt_cube, src_cube, metadata_cube])
+    iris.fileformats.netcdf.save(cube_list, filename)
+
+
+def load_regridder(filename):
+    """
+    Load a regridder scheme instance.
+
+    Loads either a `GridToMeshESMFRegridder` or a `MeshToGridESMFRegridder`.
+
+    Parameters
+    ----------
+    filename : str
+        The file name to load from.
+    """
+    with PARSE_UGRID_ON_LOAD.context():
+        cubes = iris.load(filename)
+
+    src_name = "regridder source field"
+    tgt_name = "regridder target field"
+    metadata_name = "regridder weights and metadata"
+
+    # Extract the source, target and metadata information.
+    src_cube = cubes.extract_cube(src_name)
+    tgt_cube = cubes.extract_cube(tgt_name)
+    metadata_cube = cubes.extract_cube(metadata_name)
+
+    # Determine the regridder type.
+    regridder_type = metadata_cube.attributes["regridder type"]
+    if regridder_type == "grid to mesh":
+        scheme = GridToMeshESMFRegridder
+    elif regridder_type == "mesh to grid":
+        scheme = MeshToGridESMFRegridder
+    else:
+        raise ValueError(
+            f"expected a regridder type to be either 'grid to mesh' "
```

Comment:
@stephenworsley You can turn `codecov` patch warnings off with `patch: off` in the `codecov.yml`, see https://github.com/SciTools/cf-units/blob/main/codecov.yml

I found `codecov` patching an annoyance, particularly as it causes CI to fail